### PR TITLE
Neural net

### PR DESCRIPTION
**Pull request by ElieKadoche**
Neural net training!

### CODE REVIEW

File: train_net.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        )
+
+        model.train()
+        running_loss = 0.0
+
+        for i_batch, batch in t:
+            x, y = batch[0].to(device), batch[1].to(device)
+            optimizer.zero_grad()
+            y_pred = model(x)
+            loss = criterion(torch.log10(y_pred), y)
```
Comment (XanX3601) [inline train_net.py]: You sure about log10 ?

File: train_net.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            t.set_description(
+                "Epoch {:03d} / {} - Batch loss = {:.9f}".format(
+                    epoch + 1, args.epochs, loss.item()))
+            loss.backward()
+            optimizer.step()
+
+            if i_batch == last_batch_index:
+                t.set_postfix_str("mean_total_loss = {:.9f}".format(
+                    running_loss / (last_batch_index + 1)))
+
```
Comment (XanX3601) [inline train_net.py:84]: So no test for the moment ? And no validation / cross-validation either ?

File: utils/dataset.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    5: "guard_rail",
+    6: "traffic_sign",
+    7: "traffic_light",
+    8: "pole",
+    9: "misc",
+    10: "truc",
+    11: "car",
+    12: "van",
+    13: "none",
+}
```
Comment (XanX3601) [inline utils/dataset.py:33]: Nice touch

File: utils/neural_net.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+        Cubes are considered of size 20 x 20 x 20 and
+        as described in the paper, we use 20 filters.
+        """
+        super(Net, self).__init__()
+
+        # Output classes
+        self.o = 14
+
+        # Number of channels (3 for RGB)
```
Comment (XanX3601) [inline utils/neural_net.py]:
```suggestion
        # Number of channels (4 for RGB)
```

Comment (XanX3601) [reply utils/neural_net.py]: We need 1 channel for the points and 3 for the colors

**Review by XanX3601 — CHANGES_REQUESTED** [review]
Good job !
Two or three details to discuss but it's good

**Comment by XanX3601** [comment]
Trying to reference a commit
https://github.com/XanX3601/point_cloud_labeling/pull/1/commits/79446bbbc9e2083b01f0e7d4c72c4585d6e3613c

File: train_net.py (hunk 1 above)
Comment (ElieKadoche) [reply train_net.py]: You are right. Corrected in https://github.com/XanX3601/point_cloud_labeling/pull/1/commits/bd1139d46f89b80fbe5093555d344888aead1d4a.

File: train_net.py (hunk 2 above)
Comment (ElieKadoche) [reply train_net.py:84]: We could. But to begin with, we can only test the global algorithm.

File: utils/dataset.py (hunk 3 above)
Comment (ElieKadoche) [reply utils/dataset.py:33]: Thanks

File: utils/neural_net.py (hunk 4 above)
Comment (ElieKadoche) [reply utils/neural_net.py]: Absolutely!

**Pull request merged**